src/client/src/consumer.rs

```diff
@@ -564,38 +571,159 @@ static MAX_FETCH_BYTES: Lazy<i32> = Lazy::new(|| {
     max_bytes
 });
 
+#[derive(Debug, Clone)]
+pub struct SmartStreamModule {
```

Comment:
Instead of creating another wrapper, rather use existing fluvio data type `RecordData` instead of bytes.  This will also come in handy when storing WASM bytes in the Fluvio later on.

Reply:
I think it would not be a good idea to overload `RecordData` to mean two different things. I also think that it is actually a good idea to have this `SmartStreamModule` type because we can expand on it in the future, e.g. validate that it is correct WASM before instantiating it. If you would prefer not to have a new type for this right now, I can turn it back into a `Vec<u8>`, but I think using `RecordData` is a bad idea.